scene/3d/physics/soft_body_3d.cpp

```diff
@@ -71,15 +75,70 @@ void SoftBodyRenderingServerHandler::clear() {
 }
 
 void SoftBodyRenderingServerHandler::open() {
-	write_buffer = buffer.ptrw();
+	write_buffer = buffer_curr.ptrw();
 }
 
 void SoftBodyRenderingServerHandler::close() {
 	write_buffer = nullptr;
 }
 
-void SoftBodyRenderingServerHandler::commit_changes() {
-	RS::get_singleton()->mesh_surface_update_vertex_region(mesh, surface, 0, buffer);
+void SoftBodyRenderingServerHandler::fti_pump() {
+	if (buffer_prev.is_empty()) {
+		buffer_prev.resize(buffer_curr.size());
+	}
+	SWAP(buffer_prev, buffer_curr);
+	aabb_prev = aabb_curr;
+}
+
+void SoftBodyRenderingServerHandler::commit_changes(real_t p_interpolation_fraction) {
+	real_t f = p_interpolation_fraction;
+	AABB aabb_interp = aabb_curr;
+
+	if (buffer_interp.is_empty()) {
+		buffer_interp.resize(buffer_curr.size());
+	}
+
+	if (p_interpolation_fraction < 1) {
+		// AABB.
+		if (aabb_prev != AABB() && aabb_curr != AABB()) {
+			aabb_interp = AABB(aabb_prev.position.lerp(aabb_curr.position, f), aabb_prev.size.lerp(aabb_curr.size, f));
+		}
+
+		const float *vertex_prev = reinterpret_cast<const float *>(buffer_prev.ptr() + offset_vertices);
+		const float *vertex_curr = reinterpret_cast<const float *>(buffer_curr.ptr() + offset_vertices);
+		float *vertex_interp = reinterpret_cast<float *>(buffer_interp.ptrw() + offset_vertices);
+
+		const uint32_t *normal_prev = reinterpret_cast<const uint32_t *>(buffer_prev.ptr() + offset_normal);
+		const uint32_t *normal_curr = reinterpret_cast<const uint32_t *>(buffer_curr.ptr() + offset_normal);
+		uint32_t *normal_interp = reinterpret_cast<uint32_t *>(buffer_interp.ptrw() + offset_normal);
+
+		for (uint32_t i = 0; i < vertex_count; i++) {
+			// Vertex.
+			vertex_interp[0] = Math::lerp(vertex_prev[0], vertex_curr[0], (float)f);
+			vertex_interp[1] = Math::lerp(vertex_prev[1], vertex_curr[1], (float)f);
+			vertex_interp[2] = Math::lerp(vertex_prev[2], vertex_curr[2], (float)f);
+
+			vertex_prev += stride / sizeof(float);
+			vertex_curr += stride / sizeof(float);
+			vertex_interp += stride / sizeof(float);
```

Comment:
Probably worth pre-calculating the strides in units, just to make it faster in debug in this tight loop.
I don't think it's less clear, especially if you call it something like `stride_units` and `normal_stride_units`.